src/Altinn.AccessManagement.Persistence/DelegationMetadataRepo.cs

```diff
@@ -0,0 +1,779 @@
+                { "coveredByUserId", delegationChange.CoveredByUserId.HasValue ? delegationChange.CoveredByUserId.Value : DBNull.Value },
+                { "coveredByPartyId", delegationChange.CoveredByPartyId.HasValue ? delegationChange.CoveredByPartyId.Value : DBNull.Value },
+                { "performedByUserId", delegationChange.PerformedByUserId.HasValue ? delegationChange.PerformedByUserId.Value : DBNull.Value },
+                { "blobStoragePolicyPath", delegationChange.BlobStoragePolicyPath },
+                { "blobStorageVersionId", delegationChange.BlobStorageVersionId }
+            };
+
+            string query = $"INSERT INTO delegation.delegationChanges(delegationChangeType, altinnAppId, offeredByPartyId, coveredByUserId, coveredByPartyId, performedByUserId, blobStoragePolicyPath, blobStorageVersionId)" +
+                $"VALUES (@delegationChangeType, @altinnAppId, @offeredByPartyId, @coveredByUserId, @coveredByPartyId, @performedByUserId, @blobStoragePolicyPath, @blobStorageVersionId)" +
+                $"RETURNING *;";
+
+            try
+            {
+                var res = await _connection.QueryAsync<DelegationChange>(new CommandDefinition(query, param, cancellationToken: cancellationToken));
+                return res.FirstOrDefault();
+            }
+            catch (Exception ex)
+            {
+                activity?.StopWithError(ex);
+                throw;
+            }
+            finally
+            {
+                _connection.Close();
+            }
+        }
+
+        private async Task<DelegationChange> InsertResourceRegistryDelegation(DelegationChange delegationChange, CancellationToken cancellationToken = default)
+        {
+            using var activity = TelemetryConfig.ActivitySource.StartActivity(ActivityKind.Client);
+            var param = new Dictionary<string, object>
+            {
+                { "delegationChangeType", delegationChange.DelegationChangeType },
+                { "altinnAppId", delegationChange.ResourceId }, // resourceregistryid ?
+                { "offeredByPartyId", delegationChange.OfferedByPartyId },
+                { "coveredByUserId", delegationChange.CoveredByUserId.HasValue ? delegationChange.CoveredByUserId.Value : DBNull.Value },
+                { "coveredByPartyId", delegationChange.CoveredByPartyId.HasValue ? delegationChange.CoveredByPartyId.Value : DBNull.Value },
+                { "performedByUserId", delegationChange.PerformedByUserId.HasValue ? delegationChange.PerformedByUserId.Value : DBNull.Value },
+                { "performedByPartyId", delegationChange.PerformedByPartyId.HasValue ? delegationChange.PerformedByUserId.Value : DBNull.Value },
+                { "blobStoragePolicyPath", delegationChange.BlobStoragePolicyPath },
+                { "blobStorageVersionId", delegationChange.BlobStorageVersionId },
+                { "delegatedTime", delegationChange.Created } // check ?
```

Comment:
DelegatedTime is currently set as:
`delegationChange.Created.HasValue ? delegationChange.Created.Value : DateTime.UtcNow);`

It's only migrated delegations from Altinn 2 which have the Created datetime already set on input. Otherwise the database should set current time (which I guess could/should have been done with having a default on the table column instead but..)

Reply:
Changed to recommended value. Added to issue to fix default value on column #724 .